cmd/agent/app/server.go

```diff
@@ -133,6 +135,31 @@ func (a *Agent) runProxyConnection(o *options.GrpcProxyAgentOptions, drainCh, st
 		}),
 	}
 	cc := o.ClientSetConfig(dialOptions...)
+
+	if o.ServerLeaseSelector != "" {
+		var k8sClient *kubernetes.Clientset
+		config, err := clientcmd.BuildConfigFromFlags("", o.KubeconfigPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load kubernetes client config: %v", err)
+		}
+		k8sClient, err = kubernetes.NewForConfig(config)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create kubernetes clientset: %v", err)
+		}
+		serverLeaseSelector, err := labels.Parse(o.ServerLeaseSelector)
+		if err != nil {
+			return nil, fmt.Errorf("invalid server count lease selector: %w", err)
+		}
+		serverLeaseCounter := agent.NewServerLeaseCounter(
+			k8sClient,
+			serverLeaseSelector,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create server lease counter: %w", err)
+		}
+		cc.ServerLeaseCounter = serverLeaseCounter
```

Comment:
lower bound to 1?

Reply:
There's already special handling to ensure the agent keeps trying to connect when the server count is 0, so I don't think bounding to 1 is needed.

Reply:
I don't agree. If we exit runProxyConnection because of a lease error, you wedge the cluster. Happy to chat more but I think if we're not careful feel are going to do some real damage to our reliability here.

Reply:
Lease counter now always returns at least 1.